airlock/policies.py

```diff
@@ -53,6 +69,12 @@ def check_can_add_file_to_request(workspace: "Workspace", relpath: UrlPath):
     if workspace.file_has_been_released(relpath):
         raise exceptions.RequestPermissionDenied("Cannot add released file to request")
 
+    workspace_status = workspace.get_workspace_file_status(relpath)
+    if workspace_status not in VALID_WORKSPACEFILE_STATES_TO_ADD:
+        raise exceptions.RequestPermissionDenied(
+            f"Cannot add file in status {workspace_status}"
+        )
```

Comment:
this block transferred as-is from its previous location, I'm doing more work on this in my other branch as I tease apart add_file from update_file, so I don't think it's worth grooming this right now.